File: level_1/크레인 인형뽑기 게임.js
function solution(board, moves) {
    let arr = []
    for(let i=0; i<moves.length; i++){
        for(let j=0; j<board[0].length; j++){
            if(board[j][moves[i]-1] != 0) {
                arr.push(board[j][moves[i]-1]);
                board[j][moves[i]-1] = 0
                break;
            } 
        }
    }
    const size = arr.length
    for(let i=0; i<arr.length-1; i++){
        if(arr[i] == arr[i+1]){
            arr.splice(i,2);
            i -= 2;
        }
    }
    return size - arr.length
}

File: level_1/크레인 인형뽑기 게임.java
import java.util.*;

class Solution {
    public int solution(int[][] board, int[] moves) {
        int answer = 0;
        Stack<Integer> stack = new Stack<>();
        for (int i = 0; i < moves.length; i++) {
            for (int j = 0; j < board.length; j++) {
                if (board[j][moves[i] - 1] != 0) {
                    int cur = board[j][moves[i] - 1];
                    if (stack.size() != 0 && stack.peek() == cur) {
                        stack.pop();
                        answer += 2;
                    } else
                        stack.push(cur);
                    board[j][moves[i] - 1] = 0;
                    break;
                }
            }
        }
        return answer;
    }
}